component/sections/portfolio/variant_a.js

```diff
@@ -1,136 +1,20 @@
 import React from "react";
 import { urlFor } from "../../../lib/sanity";
 
-function VariantA({ /* template,*/ caption, title, images, primaryButton }) {
-  const [category, setCategory] = React.useState("category_2");
+function VariantA({ caption, title, portfolios }) {
+  const [category, setCategory] = React.useState(portfolios?.[0]?.category); //set the first index category as initial value
+  let [viewProjects, setViewPortfolios] = React.useState(8);
```

Comment:
What is `8`? For cases like this, it's better if you initialize a variable with the said value `8` and use that variable for better readability. For example, `React.useState(projectsLength)`

Reply:
Also, it's better and common practice we make sure that we name `useState` destructured values as first - `name` and 2nd is prefixed with `set` and same name of first. 

Eg:

`const [viewProjects, setViewProjects] = React.useState()`

Reply:
Based on variant design, we may set the number of portfolios to initially display. Here, for variant A I set it to display 8 items on first render. Revised the line of code to use a constant variable for better readability and the function name.

Reply:
Thanks for accepting my proposal for change.